nestkernel/node_manager.cpp

```diff
@@ -633,7 +621,7 @@ NodeManager::prepare_node_( Node* n )
 {
   // Frozen nodes are initialized and calibrated, so that they
   // have ring buffers and can accept incoming spikes.
-  n->init_buffers();
```

Comment:
We still need this call, no?

Reply:
@clinssen this function is renamed to `init()`, and `init_buffers_()` is called from here.

Reply:
Ah yes, I see now! I suppose the original idea was for each model to have all its state variables in the State_/S_, and all its RingBuffers in the Buffers_/B_ member struct. Buffers_ now also seems to be home to things like the GSL solver, so I'm not sure this separation between Buffers_ and State_ still makes sense or can be made cleanly. Note that there's also Variables_ ("internal variables of the model"), the delineation of which is also somewhat unclear. Do we want to maintain Buffers_, State_ and Variables_ as separate member structs? (The fourth one being Parameters_.) This also with an eye to potential future "save-and-load" mechanisms.

Reply:
I will need to dig out the description of the P-S-B-V logic from deep archives and post it. Briefly:
- Parameters: part of the neuroscientific model, do not change with time during simulation
- State: part of the neuroscientific model, evolve with time during simulation
- Buffers: implementation details, persist from one call to Simulate to the next
- Variables: auxiliary values precomputed for efficiency
When introduced, this separation significantly improved our ability to think systematically about the different parts of the model, so we should make changes only with great care.

Reply:
> I will need to dig out the description of the P-S-B-V logic from deep archives and post it. Briefly:
> 
>     * Parameters: part of the neuroscientific model, do not change with time during simulation
> 
>     * State: part of the neuroscientific model, evolve with time during simulation
> 
>     * Buffers: implementation details, persist from one call to Simulate to the next
> 
>     * Variables: auxiliary values precomputed for efficiency
> 
> 
> When introduced, this separation significantly improved our ability to think systematically about the different parts of the model, so we should make changes only with great care.

@heplesser: what about the last sentence in your comment—should this PR's changes still be discussed in this context?

Reply:
@clinssen I am not entirely sure I understand what you mean by "still be discussed in this context"? My suggestion is that within this PR, we clean up in the sense of making sure that both `init_state_()` and `init_buffers_()` are called once on each node, via `node::init()`. Any further discussion should be moved to a follow-up issue.

Reply:
@heplesser @clinssen I have created issue #2050 for further discussion.